src/lxc/start.c

```diff
@@ -228,7 +231,11 @@ int lxc_check_inherited(struct lxc_conf *conf, bool closeall, int fd_to_ignore)
 			continue;
 		}
 
-		if (fd == fddir || fd == lxc_log_fd || fd == fd_to_ignore)
+		for (i = 0; i < len_fds; i++)
+			if (fds_to_ignore[i] == fd)
+				break;
+
+		if (fd == fddir || fd == lxc_log_fd || fd == fds_to_ignore[i])
```

Comment:
If the program does not hit the line `236`,  `i` will be equal to `len_fds`, and `fd == fds_to_ignore[i]`  in line `238` will access out of bounds. 

```
if (fd == fddir || fd == lxc_log_fd || ( i < len_fds && fd == fds_to_ignore[i]))
```